crates/dids/src/method/dht/mod.rs

```diff
@@ -0,0 +1,126 @@
+mod bep44;
+mod convert;
+pub mod registered_types;
+
+use std::sync::Arc;
+
+use crypto::Curve;
+use keys::key_manager::KeyManager;
+use reqwest::blocking::Client;
+
+use crate::{
+    bearer::BearerDid,
+    document::{Document, Service, VerificationMethod},
+    identifier::Identifier,
+};
+
+use self::{
+    bep44::encode_bep44_message, convert::document_packet::document_to_packet,
+    registered_types::RegisteredDidType,
+};
+
+use super::{Create, MethodError};
+
+const DEFAULT_GATEWAY_URL: &str = "https://diddht.tbddev.org";
+
+/// Concrete implementation for a did:dht DID
+pub struct DidDht;
+
+/// Options that can be used to create a did:dht DID
+pub struct DidDhtCreateOptions {
+    pub publish: bool,
+    // TODO: gatewayUri - default 'https://diddht.tbddev.org'
+    pub also_known_as: Option<Vec<String>>,
+    pub controller: Option<Vec<String>>,
+    pub service: Option<Vec<Service>>,
+    pub registered_type: Option<Vec<RegisteredDidType>>,
+    pub verification_methods: Option<Vec<VerificationMethod>>,
+}
+
+impl Create<DidDhtCreateOptions> for DidDht {
+    fn create(
+        key_manager: Arc<dyn KeyManager>,
+        options: DidDhtCreateOptions,
+    ) -> Result<BearerDid, MethodError> {
+        // Generate private and public keypair
+        let key_alias = key_manager.generate_private_key(
+            Curve::Ed25519,
+            Some("0".to_string()),
+        )?;
+        let public_key = key_manager.get_public_key(&key_alias)?;
+
+        // Create did uri
+        let public_jwk = public_key.jwk()?;
+        let jwk_string = serde_json::to_string(public_jwk.as_ref()).map_err(|_| {
+            MethodError::DidCreationFailure("failed to serialize public jwk".to_string())
+        })?;
+        let suffix = zbase32::encode_full_bytes(jwk_string.as_bytes());
+        let uri = format!("did:dht:{}", suffix);
+        let identifier = Identifier::parse(&uri).map_err(|e| {
+            MethodError::DidCreationFailure(format!("Failed to parse did:jwk uri {} {}", &uri, e))
+        })?;
+
+        // Create did document
+        let mut verification_methods = vec![VerificationMethod {
+            id: format!("{}#0", uri),
+            r#type: "JsonWebKey".to_string(),
+            controller: uri.clone(),
+            public_key_jwk: public_jwk.as_ref().clone(),
+        }];
+
+        if let Some(vms) = options.verification_methods {
+            verification_methods.extend(vms);
+        }
+
+        let document = Document {
+            id: uri.clone(),
+            verification_method: verification_methods,
+            service: options.service,
+            ..Default::default()
```

Comment:
[from the spec](https://did-dht.com/#identity-key)

> Identity Keys MUST have the [Verification Relationships](https://did-dht.com/#verification-relationships) Authentication, Assertion, Capability Invocation, and Capability Delegation.

we need to go ahead and create those four verification relationships in the document creation